### PR TITLE
chore: increases rule.go test coverage

### PR DESCRIPTION
**Pull request by M4tteoP**
PR meant to increase `rule.go` test coverage

### CODE REVIEW

File: internal/corazawaf/rule_test.go [hunk 1]
```diff
@@ -4,13 +4,187 @@
...
+		t.Errorf("Expected 0 matchdata, got %d", len(matchdata))
+	}
+	if tx.interruption != nil {
+		t.Errorf("Expected interruption not triggered because of RemoveRuleTargetByID")
+	}
+}
+
+type dummyFlowAction struct{}
+
+func (a *dummyFlowAction) Init(_ plugintypes.RuleMetadata, data string) error {
```
Comment (jcchavezs) [inline internal/corazawaf/rule_test.go:90]:
```suggestion
func (a *dummyFlowAction) Init(_ plugintypes.RuleMetadata, _ string) error {
```

File: internal/corazawaf/rule_test.go [hunk 2]
```diff
@@ -4,13 +4,187 @@
...
+	return nil
+}
+func (a *dummyFlowAction) Evaluate(r plugintypes.RuleMetadata, tx plugintypes.TransactionState) {
```
Comment (jcchavezs) [inline internal/corazawaf/rule_test.go:93]: You don´t need the receiver here (namely `a`) as you aren't accesing any field.

**Review by jcchavezs — COMMENTED** [review]
Just very small nits around parameter name and receivers as they aren't needed but overall looks great.

**Review by jcchavezs — APPROVED** [review]

**Review by fzipi — APPROVED** [review]
LGTM

**Pull request merged**